packages/app-check/src/internal-api.test.ts

```diff
@@ -171,6 +175,26 @@ describe('internal api', () => {
       errorStub.restore();
     });
 
+    it('resolves with a dummy token and an error if recaptcha failed', async () => {
+      const errorStub = stub(console, 'error');
+      const appCheck = initializeAppCheck(app, {
+        provider: new ReCaptchaV3Provider(FAKE_SITE_KEY)
+      });
+
+      const reCAPTCHASpy = stubGetRecaptchaToken('', false);
```

Comment:
Should the first parameter here, the token parameter, be related to `formatDummyToken(defaultTokenErrorData)`?

Reply:
No, that's the recaptcha token.  The flow in App Check (when using recaptcha, the default attestation provider) is
1) The client requests a recaptcha token
2) recaptcha returns a recaptcha token
3) The client takes the recaptcha token and sends it to the App Check exchange endpoint
4) The App Check exchange endpoint returns an App Check token

In this test, step 3 is being interrupted, the recaptcha token was no good (it was `''` as stubbed here), so it won't bother to send anything to the exchange endpoint and will just create a dummy App Check token locally, which is returned by `getToken()` (the function that covers the whole exchange logic).

Reply:
Ahhh. thanks so much for the detailed response!